handlers.go

```diff
@@ -432,6 +443,130 @@ func (s *Server) handleStats(c echo.Context, u *User) error {
+				}
+			}
+			multiAddrs = append(multiAddrs, a)
+		}
+
+		peerParamId, _ := peer.Decode(peerParam.ID)
+		//	add the peer(s)
+		s.Node.Peering.AddPeer(
+			peer.AddrInfo{
+				peerParamId,
+				multiAddrs,
+			})
+	}
+	return c.JSON(200, params)
+}
+
+// handlePeeringPeersRemove godoc
+// @Summary      Remove peers on Peering Service
+// @Description  This endpoint can be used to remove a Peer from the Peering Service
+func (s *Server) handlePeeringPeersRemove(c echo.Context) error {
+	var params []peer.ID
+
+	type PeeringPeerRemoveMessage struct {
+		Message     string    `json: Message`
+		PeersRemove []peer.ID `json: Peers`
+	}
+
+	if err := c.Bind(&params); err != nil {
+		return &util.HttpError{
+			Code:    400,
+			Message: util.ERR_PEERING_PEERS_REMOVE_ERROR,
+		}
+	}
+
+	for _, peerId := range params {
+		s.Node.Peering.RemovePeer(peerId)
+	}
+
+	return c.JSON(200, PeeringPeerRemoveMessage{"Removed the following Peers from Peering", params})
+}
+
+// handlePeeringPeersList godoc
+// @Summary      List all Peering peers
+// @Description  This endpoint can be used to list all peers on Peering Service
+func (s *Server) handlePeeringPeersList(c echo.Context) error {
+	var connectionCheck []peering.PeeringPeer
+	for _, peerAddrInfo := range s.Node.Peering.ListPeers() {
+
+		var peerAddrInfoAddrsStr []string
+		for _, addrInfo := range peerAddrInfo.Addrs {
+			peerAddrInfoAddrsStr = append(peerAddrInfoAddrsStr, addrInfo.String())
+		}
+		connectionCheck = append(connectionCheck, peering.PeeringPeer{
+			ID:        peerAddrInfo.ID.Pretty(),
+			Addrs:     peerAddrInfoAddrsStr,
+			Connected: (s.Node.Host.Network().Connectedness(peerAddrInfo.ID) == network.Connected),
+		})
+	}
+	return c.JSON(200, connectionCheck)
+}
+
+// handlePeeringStart godoc
+// @Summary      Start Peering
+// @Description  This endpoint can be used to start the Peering Service
+func (s *Server) handlePeeringStart(c echo.Context) error {
+	type GenericResponse struct {
+		Message string `json: "Message"`
+	}
+	err := s.Node.Peering.Start()
+	if err != nil {
+		fmt.Println(err)
+		return &util.HttpError{
+			Code:    400,
+			Message: util.ERR_PEERING_PEERS_START_ERROR,
+		}
+	}
+	return c.JSON(200, GenericResponse{Message: "Peering Started."})
+}
+
+// handlePeeringStop godoc
+// @Summary      Stop Peering
+// @Description  This endpoint can be used to stop the Peering Service
+func (s *Server) handlePeeringStop(c echo.Context) error {
+	type GenericResponse struct {
+		Message string `json: "Message"`
+	}
+	err := s.Node.Peering.Stop()
+	if err != nil {
+		return &util.HttpError{
+			Code:    400,
+			Message: util.ERR_PEERING_PEERS_STOP_ERROR,
+		}
+	}
+	return c.JSON(200, GenericResponse{Message: "Peering Stopped."})
+}
+
+// handlePeeringStatus godoc
+// @Summary      Check Peering Status
+// @Description  This endpoint can be used to check the Peering status
+func (s *Server) handlePeeringStatus(c echo.Context) error {
```

Comment:
how is this endpoint supposed to work?

Reply:
https://github.com/ipfs/go-ipfs/pull/9030

Reply:
This is approved so I'll wait for this to get merged and grab the latest version of IPFS to use the new State() function.

Reply:
when will this be updated

Reply:
No dates yet but should be soon, I can wait for the next release and update this PR once we have the new release published.